libs/client-api/src/v2/actor.rs

```diff
@@ -305,16 +324,26 @@ impl WorkspaceControllerActor {
       object_id,
       flags,
       update,
+      collab_type,
       ..
     } = &msg
     {
       let rid = source.into();
+      let changed_collab = ChangedCollab {
+        id: *object_id,
+        collab_type: *collab_type,
+      };
+      // Only insert if not already present
+      if !self.latest_changed_collabs.contains(&changed_collab) {
+        self.latest_changed_collabs.insert(changed_collab);
```

Comment:
Why do you need to check if it's there before insertion?

Reply:
```
if !self.latest_changed_collabs.contains(object_id) {
        self.latest_changed_collabs.insert(ChangedCollab {
          id: *object_id,
          collab_type: *collab_type,
        });
      }
```
Contains is cheaper than insert